source/parser.hera

```diff
@@ -925,7 +925,8 @@ CallExpressionRest
 
 # NOTE: Added shorthand x?(3) -> x?.(3)
 OptionalShorthand
-  InlineComment* QuestionMark OptionalDot ->
+  # perf: assertion to exit early
+  /(?=[\/?!])/ InlineComment* QuestionMark OptionalDot ->
```

Comment:
`!` isn't needed here, unless we place this in a rule above (which might help?).